src/modules/universal-discovery/components/finder/finder.tree.leaf.component.js

```diff
@@ -88,16 +76,21 @@ export default class FinderTreeLeafComponent extends Component {
     }
 
     render() {
-        const location = this.props.location;
-        const isForcedLocation = this.props.allowedLocations.length === 1;
+        const { location, selected, isSelectable, isLoadingChildren, allowedLocations } = this.props;
+        const isForcedLocation = allowedLocations.length === 1;
         const componentClassName = 'c-finder-tree-leaf';
-        const isSelected = this.state.selected ? `${componentClassName}--selected` : '';
-        const isNotSelectable = !this.props.isSelectable || isForcedLocation ? `${componentClassName}--not-selectable` : '';
-        const hasChildren = location.childCount ? `${componentClassName}--has-children` : '';
-        const isLoadingChildren = this.state.isLoadingChildren ? `${componentClassName}--loading` : '';
-        const finalClassName = `${componentClassName} ${isSelected} ${hasChildren} ${isLoadingChildren} ${isNotSelectable}`;
+        const isSelectedClassName = selected ? `${componentClassName}--selected` : '';
+        const isNotSelectableClassName = !isSelectable || isForcedLocation ? `${componentClassName}--not-selectable` : '';
+        const hasChildrenClassName = location.childCount ? `${componentClassName}--has-children` : '';
+        const isLoadingChildrenClassName = isLoadingChildren ? `${componentClassName}--loading` : '';
         const attrs = {
-            className: finalClassName,
+            className: [
+                componentClassName,
+                isSelectedClassName,
+                hasChildrenClassName,
+                isLoadingChildrenClassName,
+                isNotSelectableClassName,
+            ].join(' '),
```

Comment:
👍

Reply:
I would go for helper function accepting object `{ 'some-class-name': true | false }`.